private/bufpkg/bufimage/bufimagebuild/builder.go

```diff
@@ -249,8 +249,13 @@ func (b *builder) warnInvalidImports(
 			)
 			if workspaceModule, ok := workspaceFilesToModule[importFilePath]; ok {
 				if workspaceModule == "" {
-					warnMsg += ", but is found in a local workspace unnamed module. Name the module and declare it in the deps key in buf.yaml."
+					// If dependency comes from an unnamed module, that is probably a local dependency, and
+					// that module won't be pushed to the BSR. We can skip this warning.
+					continue
 				} else {
```

Comment:
For readability, it would be nice to remove the else/nesting here since we continue above.